crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -18,83 +18,131 @@ pub(super) fn place_comment<'a>(
+                handle_trailing_binary_expression_left_or_operator_comment(
+                    comment,
+                    binary_expression,
+                    locator,
+                )
+            }
+            AnyNodeRef::Keyword(_) => handle_dict_unpacking_comment(comment, locator),
+            AnyNodeRef::ExprDict(_) => handle_dict_unpacking_comment(comment, locator)
+                .or_else(|comment| handle_bracketed_end_of_line_comment(comment, locator)),
+            AnyNodeRef::ExprIfExp(expr_if) => handle_expr_if_comment(comment, expr_if, locator),
+            AnyNodeRef::ExprSlice(expr_slice) => {
+                handle_slice_comments(comment, expr_slice, locator)
+            }
+            AnyNodeRef::ExprStarred(starred) => {
+                handle_trailing_expression_starred_star_end_of_line_comment(comment, starred)
+            }
+            AnyNodeRef::ExprSubscript(expr_subscript) => {
+                if let Expr::Slice(expr_slice) = expr_subscript.slice.as_ref() {
+                    handle_slice_comments(comment, expr_slice, locator)
+                } else {
+                    CommentPlacement::Default(comment)
+                }
+            }
+            AnyNodeRef::ModModule(_) => {
+                handle_module_level_own_line_comment_before_class_or_function_comment(
+                    comment, locator,
+                )
+            }
+            AnyNodeRef::WithItem(_) => handle_with_item_comment(comment, locator),
+            AnyNodeRef::StmtFunctionDef(function_def) => {
+                handle_leading_function_with_decorators_comment(comment)
+                    .or_else(|comment| handle_leading_returns_comment(comment, function_def))
+            }
+            AnyNodeRef::StmtClassDef(class_def) => {
+                handle_leading_class_with_decorators_comment(comment, class_def)
+            }
+            AnyNodeRef::StmtImportFrom(import_from) => {
+                handle_import_from_comment(comment, import_from)
+            }
+            AnyNodeRef::ExprConstant(_) => {
+                if let Some(AnyNodeRef::ExprFString(fstring)) = comment.enclosing_parent() {
+                    CommentPlacement::dangling(fstring, comment)
+                } else {
+                    CommentPlacement::Default(comment)
+                }
+            }
+            AnyNodeRef::ExprList(_)
+            | AnyNodeRef::ExprSet(_)
+            | AnyNodeRef::ExprGeneratorExp(_)
+            | AnyNodeRef::ExprListComp(_)
+            | AnyNodeRef::ExprSetComp(_)
+            | AnyNodeRef::ExprDictComp(_)
+            | AnyNodeRef::ExprTuple(_) => handle_bracketed_end_of_line_comment(comment, locator),
+            _ => CommentPlacement::Default(comment),
+        })
+}
+
+fn handle_parenthesized_comment<'a>(
+    comment: DecoratedComment<'a>,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    let Some(preceding) = comment.preceding_node() else {
```

Comment:
Nit: We could restrict this to only run if `preceding` or `following` is an expression (or, because we have cases where the if header is the preceding, that is then followed by the first body statement)